src/buildHooks.ts

```diff
@@ -67,7 +83,51 @@ export function buildHooks<Definitions extends EndpointDefinitions>({
   mutationSelectors: MutationResultSelectors<Definitions, any>;
   mutationActions: MutationActions<Definitions>;
 }) {
-  return { buildQueryHook, buildMutationHook };
+  return { buildQueryHook, buildMutationHook, usePrefetch };
+
+  function usePrefetch<EndpointName extends QueryKeys<Definitions>>(
+    endpointName: EndpointName,
+    options?: PrefetchOptions
+  ) {
+    const dispatch = useDispatch<ThunkDispatch<any, any, AnyAction>>();
+    const store = useStore();
+    const latestStateRef = useRef(store.getState());
+
+    useLayoutEffect(() => {
+      return store.subscribe(() => {
+        latestStateRef.current = store.getState();
+      });
+    }, [store]);
+
+    return useCallback(
```

Comment:
This whole callback could also be implemented as a thunk and thus also be available for non-hook-usage with all the fancy options